plugin/markdown/plugin.js

```diff
@@ -415,6 +428,73 @@ const Plugin = () => {
 
 	}
 
+	function parseFrontMatter (content, options) {
+		options = getSlidifyOptions( options)
+
+		const parsedFrontMatter = fm(content)
+
+		content = parsedFrontMatter.body;
+		if (parsedFrontMatter.frontmatter){
+			options.metadata = yaml.load(parsedFrontMatter.frontmatter);
+			if (!('slideType' in options.metadata)) {
+				content = `Missing "slideType" in default metadata`
+				console.error(content)
+				delete options.metadata
+			}
+		}
+		return [content, options];
+	}
+
+	function parseMarkdown (markdown, options) {
+		const yamlRegex =  /```(yaml|yml)\n([\s\S]*?)```(\n[\s\S]*)?/g;
+		if (yamlRegex.test(markdown)){
+			yamlRegex.lastIndex = 0;
+
+			const markdownParts = yamlRegex.exec(markdown)
+			const metadata = markdownParts[2] || {}
+			markdown = markdownParts[3] || ''
+			if (metadata){
+				try {
+					const metadataYAML = yaml.load(metadata);
+					if (metadataYAML === undefined || metadataYAML.slideType === undefined) {
+						throw new Error("The inline metadata is not valid.")
+					}
+					options.metadata = {...options.metadata, ...metadataYAML}
+					options.attributes = 'class=' + options.metadata.slideType;
+				} catch (error) {
+					markdown = error.message
+					console.error(error)
+				}
+			}
+		} else if (options.metadata){
+			options.attributes = 'class=' + options.metadata.slideType;
+		}
+
+		return [markdown, options]
+	}
+
+	function renderTemplate(content, options) {
+		try {
+			options = getSlidifyOptions(options)
+			const url = new URL(import.meta.url);
+			const templatePath = `${url.origin}/templates/${options.metadata.slideType}-template.html`
+			const xhr = new XMLHttpRequest()
+			xhr.open('GET', templatePath, false)
+			xhr.send()
+			let tempDiv = document.createElement('div');
+			if (xhr.status === 200) {
+				tempDiv.innerHTML = Mustache.render(xhr.responseText, { content: content, metadata: options.metadata });
+			} else {
+				tempDiv.innerHTML = `Template for slideType "${options.metadata.slideType}" not found.`
+				console.error('Failed to fetch template. Status: ' + xhr.status);
```

Comment:
either template literal or string concatenation
```suggestion
				tempDiv.innerHTML = `Template for slideType "${options.metadata.slideType}" not found.`
				console.error(`Failed to fetch template. Status:  ${xhr.status}`);
```